### DISCUSSION TITLE
Resource is confused about which Panel is belongs to?

### DISCUSSION THREAD

**Discussion by donnyxray** [filamentphp/filament#7254] [General]
Using v3, beta17.

1. Create two panels, PanelA and PanelB. Set PanelA as the default using ```->default()``` in PanelAPanelProvider.
2. Create a resource, XResource, and assign it to PanelB using the ```php artisan make:filament-resource``` command.

The navigation for XResource will now appear in PanelB, as expected.

XResource::getUrl('index') however (when testing) will throw an error referring to PanelA:

```Route [filament.panel-a.resources.x.index] not defined.```

**Comment by danharrin** [comment]
`XResource::getUrl('index', panel: 'PanelB')`

Resources can be registered to multiple panels, so this needs to be passed manually

**Reply by donnyxray** [reply]
Ah! Understood. Thank you. It works now.

**Reply by discoverlance-com** [reply]
@danharrin, I am experiencing a similar issue but in this case, it's when I am using the Livewire for testing and not the direct resource url. How do I achieve something similar to `XResource::getUrl('index', panel: 'PanelB')`.

It looks like it's referencing the default admin panel which the test does not belong to so it's failing.
```php
// this example using Livewire test
 Livewire::test(PostResource\Pages\ListPosts::class)
            ->assertCanSeeTableRecords($posts)
            ->assertCountTableRecords(5);

```

**Reply by danharrin** [reply]
You can set the current panel before running the test, `Filament::setCurrentPanel(Filament::getPanel('PanelB'))`

**Reply by discoverlance-com** [reply]
@danharrin  That worked, thanks very much. I do that in a `beforeEach` hook. Will this be good to add to the documentation under panel builder testing to account for multiple panels?
